### DISCUSSION TITLE
notification not received from own server but perfectly coming from firebase console in android only

### DISCUSSION THREAD

**Discussion by saurabh874** [invertase/react-native-firebase#4245] [Q&A] [answered]
Hi,
I am using react-native-firebase (5.6.0)
react-native(0.61.4)
in android notification not received from own server but if i am trying to send dummy notification its perfectly working and displaying in notification try.
can anyone please give suggestion? 
Thanks in advance. 👍 

this is my code-
notificationListener = firebase.notifications().onNotification((notification) => {
        if (androidPlatform) {
            if (!this.getCurrentConversationState(notification)) {
                const localNotification = new firebase.notifications.Notification({
                    sound: 'newmessage.wav',
                    show_in_foreground: true,
                })
                    .setNotificationId(notification.notificationId)
                    .setTitle(this.props.translate(`EVENT.${notification.data.event_type}`))
                    .setBody(notification.body)
                    .setData({
                        title: this.props.translate(`EVENT.${notification.data.event_type}`),
                        body: notification.body,
                        show_in_foreground: true,
                        channel: notification.data.channel,
                        conversation_id: notification.data.conversation_id,
                        sender_type: notification.data.sender_type,
                        campaign_id: notification.data.campaign_id,
                        event_type: notification.data.event_type,
                        sender_id: notification.data.sender_id,
                    })
                    .android.setChannelId('GefenNotifications') // e.g. the id you chose above
                    // todo
                    .android.setSmallIcon('ic_notification') // create this icon in Android Studio
                    .android.setColor('#4D7DA3') // you can set a color here
                    .android.setPriority(firebase.notifications.Android.Priority.High);
                console.log('localNotification', localNotification);
                firebase.notifications()
                    .displayNotification(localNotification)
                    .catch(err => console.error(err));
            }
        } else {
            console.log('notification_forground', notification);
            if (!this.getCurrentConversationState(notification)) {
                const localNotification = new firebase.notifications.Notification({
                    sound: 'newmessage.wav',
                    show_in_foreground: true,
                })
                    .setNotificationId(notification.notificationId)
                    .setTitle(this.props.translate(`EVENT.${notification.data.event_type}`))
                    .setBody(notification.body)
                    .setData({
                        title: this.props.translate(`EVENT.${notification.data.event_type}`),
                        subtitle: notification.subtitle,
                        body: notification.body,
                        show_in_foreground: true,
                        channel: notification.data.channel,
                        conversation_id: notification.data.conversation_id,
                        sender_type: notification.data.sender_type,
                        campaign_id: notification.data.campaign_id,
                        event_type: notification.data.event_type,
                        sender_id: notification.data.sender_id,
                    })
                    .ios.setBadge(notification.ios.badge);

                firebase.notifications()
                    .displayNotification(localNotification)
                    .catch(err => console.error(err));
            }
        }
    })

**Comment by mikehardy** [comment, marked as answer]
v5 of react-native-firebase is well out of support. The messaging package has received a great deal of bug fixes since then, which you will want to make things work reliably. You need to upgrade to current stable versions then attempt to reproduce

https://rnfirebase.io/migrating-to-v6

**Reply by saurabh874** [reply]
@mikehardy  absolutely agree with your point but  if i will handle notification from V6  is it necessary to add react-native-push-notification or some other 3rd party for display notification because in doc they have mentioned.

**Reply by mikehardy** [reply]
...and that's the future. It's time
